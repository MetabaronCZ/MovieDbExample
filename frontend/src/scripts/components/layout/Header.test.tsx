import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Header } from './Header';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/layout/Header', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Header />, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const header = container.getElementsByTagName('header');
    expect(header.length).toEqual(1);

    const logo = getByText(container, 'Databáze filmů USA (1900 - 1999)');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('href', '/');
  });
});
