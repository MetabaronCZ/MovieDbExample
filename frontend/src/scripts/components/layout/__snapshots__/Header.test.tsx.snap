// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Header should render properly 1`] = `
.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

.c0 {
  padding: 0 16px;
}

.c2 {
  display: block;
  text-decoration: none;
  text-transform: uppercase;
  color: #98c3d9;
}

.c2:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1"
    >
      <a
        class="c2"
        href="/"
      >
        Databáze filmů USA (1900 - 1999)
      </a>
    </h1>
  </header>
</div>
`;
